include/openthread/ip6.h

```diff
@@ -97,6 +97,21 @@ struct otIp6Prefix
  */
 typedef struct otIp6Prefix otIp6Prefix;
 
+/**
+ * IPv6 Address types
+ *
+ */
+enum
+{
+    kLinkLocal = 0, ///< Link-local address
+    kSlaac     = 1, ///< SLAAC assigned address
+    kDhcpv6    = 2, ///< DHCPv6 assigned address
+    kManual    = 3, ///< Manually assigned address
+    kAloc      = 4, ///< Aloc address
+    kMeshLocal = 5, ///< Mesh-local address
+    kDua       = 6, ///< DUA address
+};
```

Comment:
Since this is public OT API/definition, we need to follow C style constants (capital case with `_`) and starting with prefix `OT_`. Thanks.

Reply:
Like this?
```
enum
{
    OT_ADDRESSORIGIN_SYSTEM = 0, ///< Address assigned by the system
    OT_ADDRESSORIGIN_SLAAC  = 1, ///< SLAAC assigned address
    OT_ADDRESSORIGIN_DHCPV6 = 2, ///< DHCPv6 assigned address
    OT_ADDRESSORIGIN_MANUAL = 3, ///< Manually assigned address
};
```

Reply:
Yes. Thanks. I would aslo suggest adding `_` in `ADDRESS_ORIGIN`?

Reply:
Reworked accordingly.